Update to readme document to show how to process any files with a regex (#551)
Aiven-Open/cloud-storage-connectors-for-apache-kafka: s3-source-connector/README.md

The configuration property `file.name.template` is **mandatory**. If not set **no objects will be processed**.

> From version 3.4.0 when using "object_hash" distribution type it is possible to set this to ".*" to match all object keys or use a regex to match specific object keys. When using partition distribution type you must use the placeholders described below.

Reviewer: ```suggestion
From version 3.4.0, when using the `object_hash` distribution type, you can set this to `.*` to match all object keys or use a regular expression to match specific object keys. When using the partition distribution type, you must use the placeholders described below.
```
